### PR TITLE
[@types/node] Update type definitions for `dns/promises`

### PR DESCRIPTION
**Pull request by ghost**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test node`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: #68750
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/node/dns/promises.d.ts [hunk 1]
```diff
@@ -355,6 +355,30 @@ declare module "dns/promises" {
      * @param order must be `'ipv4first'` or `'verbatim'`.
      */
     function setDefaultResultOrder(order: "ipv4first" | "verbatim"): void;
+    const NODATA: string;
```
Comment (peterblazejewicz) [inline types/node/dns/promises.d.ts:358]: IMO those are more like enumerated values (for errors), so more like:
```ts
const NODATA: "ENODATA"
```

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/node/v18/dns.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 96fdbc0ffdaf68d9a61e95c262c09e6d56f12d42
</p>

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/node/v16/dns.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3f3511f2e6e9c43cf98b6668929b9ec66be7681a
</p>

**Comment by codershiba** [comment]
@Semigradsky @peterblazejewicz Please take a look.

**Comment by codershiba** [comment]
Something seems wrong here with the tests... @jakebailey can you take a look?

**Comment by jakebailey** [comment]
I cancelled and reran it but this is GitHub's problem, not our infra. A close reopen would achieve the same thing so I would suggest that instead of pinging maintainers

**Review by sheetalkamat — APPROVED** [review]

**Comment by codershiba** [comment]
Ready to merge

**Pull request merged**